### PR TITLE
text going out of the section borders

### PR DESCRIPTION
**Pull request by yonatan-roth**
## Status

- [x] Ready

## Related Issues

Fixes https://github.com/demisto/etc/issues/44641
Fields position issue - https://github.com/demisto/etc/issues/45346 

## Description
1. Revert border logic in code for items-section.
2. Refactor items-section to grid layout instead of absolute calculations

### CODE REVIEW

**Comment by yonatan-roth** [comment]
Hey @idanizi-panw , @Bars92 please review.
@idanizi-panw notice I've reverted the custom border logic.

File: src/components/Layouts/ReportLayout.less [hunk 1]
```diff
@@ -21,16 +21,14 @@ h1 {
...
+}
 
-  &.bordered {
-    border: 1px solid #DADBDB !important;
-    padding: 10px 12px !important;
-  }
+.bordered {
+  border: 1px solid #DADBDB !important;
+  padding: 10px 12px !important;
 }
```
Comment (idanizi-panw) [inline src/components/Layouts/ReportLayout.less:32]: so were it used now if we make it available at global scope again?

File: src/components/Sections/ItemsSection.js [hunk 2]
```diff
@@ -177,46 +175,50 @@ class ItemsSection extends Component {
...
+                  const id = this.getSectionItemKey(item);
+                  const itemDisplayType = ItemsSection.getItemDisplayType(item);
+                  const mainClass = classNames('section-item', {
+                    [itemDisplayType]: true,
+                    'first-column': allItemsAreDisplayedAsCards && item.startCol === 0,
+                    'last-column': allItemsAreDisplayedAsCards && item.endCol === columns,
+                    'last-row': allItemsAreDisplayedAsCards && item.index === lastRowIndex
+                  });
+                  const applyStyle = {
+                    gridColumn: `${item.startCol + 1}/${item.endCol + 1}`
```
Comment (idanizi-panw) [inline src/components/Sections/ItemsSection.js:194]: 🎉

File: src/components/Sections/ItemsSection.js [hunk 3]
```diff
@@ -177,46 +175,50 @@ class ItemsSection extends Component {
...
-                    className={mainClass}
-                    style={applyStyle}
-                  >
-                    <div className="section-item-header" style={item.headerStyle}>
-                      {item.fieldName}
+              <div
+                className="items-container" style={{
+                  display: 'grid',
+                  gridTemplateColumns: `repeat(${columns}, 1fr)`
+              }}
```
Comment (idanizi-panw) [inline src/components/Sections/ItemsSection.js:182]: noice

File: src/components/Sections/ItemsSection.less [hunk 4]
```diff
@@ -12,7 +12,6 @@
   .section-item {
     display: flex;
     flex-direction: column;
-    position: absolute;
```
Comment (idanizi-panw) [inline src/components/Sections/ItemsSection.less:15]: needs some thorough QA tests - potentially harmful ⚠️

File: src/utils/layout.js [hunk 5]
```diff
@@ -31,11 +31,6 @@ function isPageBreakSection(section) {
       section.data && ((isString(section.data) ? section.data : section.data.text) || '').includes(PAGE_BREAK_KEY));
 }
 
-export function shouldParentSectionBeNotBordered(section) {
-  return section.displayType === SECTION_ITEMS_DISPLAY_LAYOUTS.card &&
-    section.data.length === 1 && section.data[0].fieldType === SECTION_TYPES.markdown;
-}
-
```
Comment (idanizi-panw) [inline src/utils/layout.js:38]: 👍

**Review by idanizi-panw — APPROVED** [review]
with some notes

File: src/components/Layouts/ReportLayout.less (hunk 1 above)
Comment (yonatan-roth) [reply src/components/Layouts/ReportLayout.less]: Removed the classes - unused.

**Pull request merged**